src/node/rpc/frontend.h

```diff
@@ -791,32 +708,18 @@ namespace ccf
               "No corresponding caller entry exists."),
             ctx.pack.value());
         }
-        ctx.caller_cert = caller.value().cert;
+        ctx.session.caller_cert = caller.value().cert;
       }
 
-      auto rpc = unpack_json(input, ctx.pack.value());
-      if (!rpc.first)
+      // Store client signature. It is assumed that the forwarder node has
+      // already verified the client signature.
+      if (ctx.signed_request.has_value())
```

Comment:
I think we need to wrap this with `if (consensus == nullptr || consensus->is_primary())` as, even if the command is forwarded, the primary may have changed in the meantime and the forwarded command should not write to the KV. It probably won't matter as no transaction will be committed but we never know.

Reply:
I think in that case we have to rely on the transaction failing later with an "I'm not primary" error? If we somehow _do_ execute this transaction and make changes to the kv, we should record the signature as well. More checks means more possible paths in buggy scenarios.